File: app/src/main/java/com/neocaptainnemo/cv/ui/common/CommonAdapter.java
package com.neocaptainnemo.cv.ui.common;

import android.content.Context;
import android.databinding.DataBindingUtil;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import com.neocaptainnemo.cv.R;
import com.neocaptainnemo.cv.databinding.ItemSectionBinding;
import com.neocaptainnemo.cv.model.CommonSection;
import com.neocaptainnemo.cv.ui.ArrayAdapter;

class CommonAdapter extends ArrayAdapter<CommonSection, CommonAdapter.ViewHolder> {

    CommonAdapter(Context context) {
        super(context);
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        ItemSectionBinding binding = DataBindingUtil.inflate(LayoutInflater.from(context),
                R.layout.item_section, parent, false);
        return new ViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        holder.binding.setViewModel(new CommonViewModel(data.get(position)));
    }

    static class ViewHolder extends RecyclerView.ViewHolder {

        ItemSectionBinding binding;

        ViewHolder(ItemSectionBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }
    }
}

File: app/src/main/java/com/neocaptainnemo/cv/ui/common/CommonAdapter.kt
package com.neocaptainnemo.cv.ui.common

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.neocaptainnemo.cv.R
import com.neocaptainnemo.cv.fromHtml
import com.neocaptainnemo.cv.model.CommonSection
import com.neocaptainnemo.cv.ui.ArrayAdapter
import kotlinx.android.synthetic.main.item_section.view.*

class CommonAdapter : ArrayAdapter<CommonSection, CommonAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder =
            ViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.item_section, parent, false))


    override fun onBindViewHolder(holder: ViewHolder, position: Int) {

        val item = data[position]

        with(holder.root) {
            commonTitle.text = item.title
            commonDescription.text = item.description.fromHtml()
        }
    }

    class ViewHolder(val root: View) : RecyclerView.ViewHolder(root)
}
